src/Squirrel/UpdateManager.ApplyReleases.cs

```diff
@@ -355,31 +412,52 @@ async Task<ReleaseEntry> createFullPackagesFromDeltas(IEnumerable<ReleaseEntry>
                 // release 3: 40 => 60
                 // release 4: 60 => 80
                 // release 5: 80 => 100
-                // 
-
-                // Smash together our base full package and the nearest delta
-                var ret = await Task.Run(() => {
-                    var basePkg = new ReleasePackage(Path.Combine(rootAppDirectory, "packages", currentVersion.Filename));
-                    var deltaPkg = new ReleasePackage(Path.Combine(rootAppDirectory, "packages", releasesToApply.First().Filename));
+                //
 
-                    var deltaBuilder = new DeltaPackageBuilder(Directory.GetParent(this.rootAppDirectory).FullName);
+                var basePkg = new ReleasePackage(Path.Combine(rootAppDirectory, "packages", currentVersion.Filename));
+                var packagesPath = Path.GetDirectoryName(basePkg.InputPackageFile);
+                var unpackedBasePkgPath = Path.Combine(packagesPath, basePkg.Version.ToString());
+                var deltaBuilder = new DeltaPackageBuilder(Directory.GetParent(this.rootAppDirectory).FullName);
+
+                string baseTempPath;
+                ReleasePackage finalPkgMetadata = null;
+
+                // Create a temporary directory where to copy last app version.
+                using (Utility.WithTempDirectory(out baseTempPath, Directory.GetParent(this.rootAppDirectory).FullName)) {
+                    await Task.Run(async () => {
+                        // This condition is here for the backward compatibility to be able to apply deltas to the Squirrel version 
+                        // which did multiple times packing before performance improvement (redmine #15144).
```

Comment:
So even when skipping several versions in between, the update would still work, right?

Reply:
Exactly, it does not matter on your current version or even on the update type (delta/multiple delta/full package update).